### PR TITLE
WIP: Fix issues with changed IRIs effecting test_excelparser

### PR DESCRIPTION
**Pull request by jesper-friis**
# Description
Fix issues with changed IRIs effecting test_excelparser.

Added `difference()` method to Ontology for easier debugging. I think it is worth keeping.

Note that this branch is branched out from the directory-layout in PR #696.

## Type of change

- [x] Bug fix.
- [ ] New feature.
- [ ] Documentation update.
- [ ] Test update.

## Checklist

This checklist can be used as a help for the reviewer.

- [ ] Is the code easy to read and understand?
- [ ] Are comments for humans to read, not computers to disregard?
- [ ] Does a new feature has an accompanying new test (in the CI or unit testing schemes)?
- [ ] Has the documentation been updated as necessary?
- [ ] Does this close the issue?
- [ ] Is the change limited to the issue?
- [ ] Are errors handled for all outcomes?
- [ ] Does the new feature provide new restrictions on dependencies, and if so is this documented?

### CODE REVIEW

File: ontopy/ontology.py [hunk 1]
```diff
@@ -1941,6 +1941,14 @@ def new_annotation_property(
...
         return self.new_entity(name, parent, "annotation_property")
 
+    def difference(self, other: owlready2.Ontology) -> set:
+        """Return a set of triples that are in this, but not in the
+        `other` ontology."""
+        # pylint: disable=invalid-name
+        s1 = set(self.get_unabbreviated_triples(blank="_:b"))
+        s2 = set(other.get_unabbreviated_triples(blank="_:b"))
+        return s1.difference(s2)
+
```
Comment (francescalb) [inline ontopy/ontology.py:1951]: Nice.

File: ontopy/ontology.py [hunk 2]
```diff
@@ -2008,31 +2016,31 @@ def _unabbreviate(
 
 
 def _get_unabbreviated_triples(
-    self, subject=None, predicate=None, obj=None, blank=None
+    onto, subject=None, predicate=None, obj=None, blank=None
```
Comment (francescalb) [inline ontopy/ontology.py:2019]: Well spotted!

File: tests/test_excelparser/result_ontology/fromexcelonto.ttl [hunk 3]
```diff
@@ -12,7 +12,7 @@
         "Jesper Friis"@en,
         "Sylvain Gouttebroze"@en ;
     dcterms:title "A test domain ontology"@en ;
-    owl:imports <http://emmo.info/emmo-inferred>,
+    owl:imports <http://emmo.info/emmo>,
```
Comment (francescalb) [inline tests/test_excelparser/result_ontology/fromexcelonto.ttl:15]: I do not understand why. We import beta-4 which should not have changed as development is now happening in beta5.

File: tests/test_excelparser/onto_only_classes.xlsx
Comment (francescalb) [inline tests/test_excelparser/onto_only_classes.xlsx:1]: I have tried to compare the files visually, but can't spot the change. Can you explain?

File: tests/test_excelparser/test_excelparser.py [hunk 4]
```diff
@@ -119,6 +125,12 @@ def test_excelparser_only_classes(repo_dir: "Path") -> None:
     # Used for printing new ontology when debugging
     # ontology.save("test_only_classes.ttl")
 
+    print("-----  only in onto  -----")
+    print(onto.difference(ontology))
+
+    print("-----  only in ontology  -----")
+    print(ontology.difference(onto))
+
```
Comment (francescalb) [inline tests/test_excelparser/test_excelparser.py:133]: Should be remobed

File: tests/test_excelparser/test_excelparser.py [hunk 5]
```diff
@@ -91,14 +91,20 @@ def test_excelparser(repo_dir: "Path") -> None:
...
 def test_excelparser_only_classes(repo_dir: "Path") -> None:
     """This loads the excelfile used and tests that the resulting ontology prior
     to version 0.5.2 in which only classes where considered, but with empty sheets
     for properties."""
+
+    # if True:
+    #    from pathlib import Path
+    #
+    #    repo_dir = Path(__file__).resolve().parent.parent.parent
+
```
Comment (francescalb) [inline tests/test_excelparser/test_excelparser.py:107]: Should be removed

File: tests/test_excelparser/test_excelparser.py [hunk 6]
```diff
@@ -143,5 +155,5 @@ def test_excelparser_only_classes(repo_dir: "Path") -> None:
         update_xlspath, force=True, input_ontology=ontology
     )
     assert updated_onto.ATotallyNewPattern
-    assert updated_onto.Pattern.iri == onto.Pattern.iri
+    assert updated_onto.FinitePattern.iri == onto.FinitePattern.iri
```
Comment (francescalb) [inline tests/test_excelparser/test_excelparser.py:158]: We should check both Pattern (in the imported ontology) and FinitePattern (newly created)

File: tests/test_excelparser/result_ontology/fromexcelonto.ttl (hunk 3 above)
Comment (jesper-friis) [reply tests/test_excelparser/result_ontology/fromexcelonto.ttl:15]: It is due to backward corrections of ontology IRIs on GitHub Pages. Probably a bad decision, but now it is done and should not be reverted again...

File: tests/test_excelparser/test_excelparser.py (hunk 6 above)
Comment (jesper-friis) [reply tests/test_excelparser/test_excelparser.py:156]: Pattern doesn't exists in 1.0.0-beta5, which is the one we have here.

**Review by francescalb — APPROVED** [review]
Good, but check that pylint passes.

**Pull request merged**